Quadriga/src/main/java/edu/asu/spring/quadriga/rest/ConceptCollectionRestController.java

```diff
@@ -73,425 +75,418 @@
+        httpHeaders.setContentType(MediaType.valueOf(accept));
+        return new ResponseEntity<String>(ccId, httpHeaders, HttpStatus.OK);
+    }
+
+    /**
+     * Rest interface for the getting concept details i.e, list of items in the
+     * collection http://<<URL>:
+     * <PORT>>/quadriga/rest/conceptdetails/{collectionID}
+     * http://localhost:8080/quadriga/rest/conceptdetails/167
+     * 
+     * @author SatyaSwaroop Boddu
+     * @param collectionID
+     *            {@link ConceptCollection} ID to access the object details from
+     *            the DB
+     * @param model
+     *            {@link ModelMap} object to access any user request
+     * @return Response would be the XML containing the concepts belonging to
+     *         {@link ConceptCollection} ID.
+     * @throws RestException
+     * @throws Exception
+     */
+    @RequestMapping(value = "rest/conceptdetails/{collectionID}", method = RequestMethod.GET, produces = "application/xml")
+    public ResponseEntity<String> getConceptList(@PathVariable("collectionID") String collectionID, ModelMap model,
+            HttpServletRequest req, Principal principal) throws RestException {
+        VelocityEngine engine = null;
+        Template template = null;
+        StringWriter sw = new StringWriter();
+        collection = collectionFactory.createConceptCollectionObject();
+        collection.setConceptCollectionId(collectionID);
+        try {
+            engine = restVelocityFactory.getVelocityEngine(req);
+            engine.init();
+            conceptControllerManager.fillCollectionDetails(collection, principal.getName());
+            template = engine.getTemplate("velocitytemplates/conceptdetails.vm");
+            VelocityContext context = new VelocityContext(restVelocityFactory.getVelocityContext());
+            context.put("list", collection.getConceptCollectionConcepts());
+            context.put("conceptPowerURL", conceptPowerURL);
+            context.put("path", updateConceptPowerURLPath);
+            template.merge(context, sw);
+            return new ResponseEntity<String>(sw.toString(), HttpStatus.OK);
+
+        } catch (ResourceNotFoundException e) {
+            logger.error("Exception:", e);
+            throw new RestException(404);
+        } catch (ParseErrorException e) {
+            logger.error("Exception:", e);
+            throw new RestException(404);
+        } catch (MethodInvocationException e) {
+            logger.error("Exception:", e);
+            throw new RestException(403);
+        } catch (QuadrigaStorageException e) {
+            logger.error("Exception:", e);
+            throw new RestException(500);
+        } catch (QuadrigaAccessException e) {
+            logger.error("Exception:", e);
+            throw new RestException(403);
+        } catch (Exception e) {
+            logger.error("Exception:", e);
+            throw new RestException(403);
+        }
+
+    }

```

Comment:
please check the whole file for my comments (eg. all variable names, logger output, etc).